kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,872 @@
+                    if (j != nKnotSpans[0]-1)
+                        if (knotSpansAvailable[idMatrix][i][j+1] == 0) { 
+                            Point gaussPoint = Point((j+1+0.5) * knotStepUV[0]+startingPos[0], (i+0.5) * knotStepUV[1]+startingPos[1], 0);
+                            if (IsPointInsideSkinBoundary(gaussPoint, testBin, rSkinModelPart)) {knotSpansAvailable[idMatrix][i][j+1] = 1;}
+                        }
+                    //down node
+                    if (j != 0)
+                        if (knotSpansAvailable[idMatrix][i][j-1] == 0) { 
+                            Point gaussPoint = Point((j-1+0.5) * knotStepUV[0]+startingPos[0], (i+0.5) * knotStepUV[1]+startingPos[1], 0);
+                            if (IsPointInsideSkinBoundary(gaussPoint, testBin, rSkinModelPart)) {knotSpansAvailable[idMatrix][i][j-1] = 1;}
+                        } 
+
+                    // corner right-down node
+                    if (j != 0 && i != nKnotSpans[1]-1)
+                        if (knotSpansAvailable[idMatrix][i+1][j-1] == 0) {
+                            Point gaussPoint = Point((j-1+0.5) * knotStepUV[0]+startingPos[0], (i+1+0.5) * knotStepUV[1]+startingPos[1], 0);
+                            if (IsPointInsideSkinBoundary(gaussPoint, testBin, rSkinModelPart)) {knotSpansAvailable[idMatrix][i+1][j-1] = 1;}
+                        }
+                    // corner left-down node
+                    if (j != 0 && i != 0)
+                        if (knotSpansAvailable[idMatrix][i-1][j-1] == 0) {
+                            Point gaussPoint = Point((j-1+0.5) * knotStepUV[0]+startingPos[0], (i-1+0.5) * knotStepUV[1]+startingPos[1], 0);
+                            if (IsPointInsideSkinBoundary(gaussPoint, testBin, rSkinModelPart)) {knotSpansAvailable[idMatrix][i-1][j-1] = 1;}
+                        }
+                    // corner right-up node
+                    if (j != nKnotSpans[0]-1 && i != nKnotSpans[1]-1)
+                        if (knotSpansAvailable[idMatrix][i+1][j+1] == 0) {
+                            Point gaussPoint = Point((j+1+0.5) * knotStepUV[0]+startingPos[0], (i+1+0.5) * knotStepUV[1]+startingPos[1], 0);
+                            if (IsPointInsideSkinBoundary(gaussPoint, testBin, rSkinModelPart)) {knotSpansAvailable[idMatrix][i+1][j+1] = 1;}
+                        }
+                    // corner left-up node
+                    if (j != nKnotSpans[0]-1 && i != 0)
+                        if (knotSpansAvailable[idMatrix][i-1][j+1] == 0) {
+                            Point gaussPoint = Point((j+1+0.5) * knotStepUV[0]+startingPos[0], (i-1+0.5) * knotStepUV[1]+startingPos[1], 0);
+                            if (IsPointInsideSkinBoundary(gaussPoint, testBin, rSkinModelPart)) {knotSpansAvailable[idMatrix][i-1][j+1] = 1;}
+                        }
+
+                    // Create 25 "fake" GaussPoints to check if the majority are inside or outside
+                    const int numFakeGaussPoints = 5;
```

Comment:
```suggestion
                    // Create 5 "fake" GaussPoints to check if the majority are inside or outside
                    const int num_fake_gauss_points = 5;
```
Why 5? Are you sure that this works for all cases? If not make it user-definable.